server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java

```diff
@@ -113,10 +121,73 @@ public void testPreventJoinClusterWithUnsupportedIndices() {
         Metadata metadata = metaBuilder.build();
         expectThrows(
             IllegalStateException.class,
-            () -> NodeJoinExecutor.ensureIndexCompatibility(IndexVersions.MINIMUM_COMPATIBLE, IndexVersion.current(), metadata)
+            () -> NodeJoinExecutor.ensureIndexCompatibility(
+                IndexVersions.MINIMUM_COMPATIBLE,
+                IndexVersions.MINIMUM_READONLY_COMPATIBLE,
+                IndexVersion.current(),
+                metadata
+            )
         );
     }
 
+    public void testJoinClusterWithReadOnlyCompatibleIndices() {
+        {
+            var indexMetadata = IndexMetadata.builder("searchable-snapshot")
+                .settings(
+                    Settings.builder()
+                        .put(INDEX_STORE_TYPE_SETTING.getKey(), SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                        .put(IndexMetadata.SETTING_VERSION_CREATED, IndexVersions.MINIMUM_READONLY_COMPATIBLE)
```

Comment:
Got fooled here, I wonder if you should comment (or assert) that readonly==regular compatibility to explain to reader?

Reply:
Assertion added